test/functional/JIT_Test/src/jit/test/jitt/misc/SanityCompareAndBranch.java

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package jit.test.jitt.misc;
+
+import org.testng.annotations.Test;
+import org.testng.Assert;
+
+@Test(groups = { "level.sanity", "component.jit"})
+public class SanityCompareAndBranch extends jit.test.jitt.Test {
+
+   private int testAddressCompare(Object obj1, Object obj2) {
+      int retVal = 0;
+      if (obj1.getClass() == java.lang.String.class) {
+         String temp = (String)obj1;
+         for (int i=0; i < temp.length(); i++) {
+            if (temp.charAt(i) == 'i')
+               retVal++;
+         }
+      }
+      if (obj1.getClass() == obj2.getClass())
+         retVal = 0;
+      return retVal;
+   }
+
+   @Test
+   public void testCompareAndBranch() {
+      SanityCompareAndBranch x = new SanityCompareAndBranch();
+      String a1 = "This is Demo String a1";
+      String a2 = "This is Demo String a2";
+      Assert.assertEquals(0, testAddressCompare(a1,a2));
+      String a3 = "This is Demo String a3";
+      String a4 = "This is Demo String a4";
+      Assert.assertEquals(0, testAddressCompare(a3, a4));
+   }
```

Comment:
We should add a bit of documentation on what exactly this unit test is trying to capture. Things we should do:

1. Explain what issue the test is trying to capture
2. Reference the fix

There is a decent example of this in [1] for a Z bug.

[1] https://github.com/eclipse/openj9/blob/3fb280271d4b3093c611858b62d13f9fba6c6700/test/functional/JIT_Test/src/jit/test/ra/TestOOLSpill31Bit.java#L41-L49